protocol/x/clob/keeper/mev.go

```diff
@@ -354,24 +365,33 @@ func (k Keeper) RecordMevMetrics(
 }
 
 // GetClobMetadata fetches the mid prices for all CLOB pairs and the CLOB pairs themselves.
-// This function falls back to use the oracle price if any of the mid prices are missing.
+// This function falls back to use the oracle price if any of the mid prices are missing
+// or if the spread is greater-than-or-equal-to the max spread.
 func (k Keeper) GetClobMetadata(
 	ctx sdk.Context,
 ) (
-	clobMidPrices map[types.ClobPairId]types.Subticks,
-	clobPairs map[types.ClobPairId]types.ClobPair,
+	clobMetadata map[types.ClobPairId]ClobMetadata,
 ) {
-	clobMidPrices = make(map[types.ClobPairId]types.Subticks)
-	clobPairs = make(map[types.ClobPairId]types.ClobPair)
+	clobMetadata = make(map[types.ClobPairId]ClobMetadata)
 
 	for _, clobPair := range k.GetAllClobPairs(ctx) {
 		clobPairId := clobPair.GetClobPairId()
-		var midPriceSubticks types.Subticks
 
-		// Get the mid price if it exists, otherwise get the oracle price.
-		if midPrice, exist := k.MemClob.GetMidPrice(ctx, clobPairId); exist {
-			midPriceSubticks = midPrice
-		} else {
+		midPriceSubticks, bestBid, bestAsk, exist := k.MemClob.GetMidPrice(ctx, clobPairId)
+
+		// Use the oracle price instead of the mid price if the mid price doesn't exist or
+		// the spread is greater-than-or-equal-to the max spread.
+		if !exist || new(big.Rat).SetFrac(
+			new(big.Int).SetUint64(uint64(bestAsk.Subticks-bestBid.Subticks)),
+			new(big.Int).SetUint64(uint64(bestBid.Subticks)), // Note that bestBid cannot be 0 if exist is true.
+		).Cmp(MAX_SPREAD_BEFORE_FALLING_BACK_TO_ORACLE) >= 0 {
+			telemetry.IncrCounterWithLabels(
```

Comment:
Didn't Jonathan update this file to use custom metrics functions? E.g. [here](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/keeper/mev.go#L104-L107). Maybe should update this usage up to you